Ports Dynamic Gamemode Code from Beestation.

Xoxeyos opened the pull request:
Ports over the Dynamic Gamemode Code from Beestation, I have tested it, all seems to be well. Some gamemodes are missing from Dynamic, such as gangs, but I could add that in later. Also a readme for documentation. The readme is in 'code\game\gamemodes\dynamic' .

🆑 Xoxeyos
rscadd: Ports over general dynamic code from Beestation. Thank you qwertyquerty, ShiftyRail, and Toxici11i for making this possible.
/🆑

Xoxeyos committed:
Adds support for dynamic mathing

Xoxeyos committed:
Adds Dynamic Main Files.

Xoxeyos committed:
Adds Dynamic loading to yogstation.dme

Xoxeyos committed:
Probably adds dynamic related changes.

Xoxeyos committed:
Fixes code I didn't add, won't fix my retardation.

Xoxeyos committed:
Adds "/datum/game_mode/proc/admin_panel()" because it's apparently important.

Xoxeyos committed:
Adds dynamic related code.

Xoxeyos committed:
Adds more dynamic related code.

Xoxeyos committed:
Fixes error: possible unreachable code here on line 313

Xoxeyos committed:
Fixes copy and paste error. 


code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm, line 390, column 2:
error: possible unreachable code here

Xoxeyos committed:
Adds a testing change to fix the lint

Xoxeyos committed:
Undos paranthesis

Xoxeyos committed:
LintFixTry#2

SomeguyManperson committed:
Merge branch 'master' into Dynamic#1

fluffe9911 committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

fluffe9911 reviewed (approved):
Doesn't look like its gonna crash the server or anything

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

fluffe9911 committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

fluffe9911 committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Merge branch 'master' into Dynamic#1

Xoxeyos committed:
Adds Shadowling

Xoxeyos committed:
adds a slash

SomeguyManperson committed:
adds some logging

Xoxeyos committed:
Adds IMMERSION

Thank you Theos for the help!

Xoxeyos committed:
FIXES     IMMERSION

Xoxeyos committed:
Adds Mindshield Implants for Dynamic

Xoxeyos committed:
Adds the mindshield implant into dynamic

Xoxeyos committed:
Adds the mindshield stuff yadayadayda

Xoxeyos committed:
adds another slash

Xoxeyos committed:
removes said slash

Xoxeyos committed:
test addes imp_ before mindshield on line 427

Xoxeyos committed:
redo

Xoxeyos committed:
test

Xoxeyos committed:
Going to try and do this again #1

Xoxeyos committed:
syndie_kit/mindshield/

Xoxeyos committed:
gets rid of an error

Xoxeyos committed:
renames to imp_mindshield

Xoxeyos committed:
Maybe I'm just super retarded.

Xoxeyos committed:
v

Xoxeyos committed:
#include "code\game\objects\items\implants\implant_mindshieldtot.dm"

DOES THIS FIX IT DOES THIS FIX IT

Xoxeyos committed:
oops

Xoxeyos committed:
recompile

Xoxeyos committed:
Part of getting rid of weights, relies on dice rolls

Xoxeyos committed:
Adds dice roll, adds the ability for clock cult to roll

Xoxeyos committed:
Readjusts weight and costs, no more weight

Xoxeyos committed:
ahem fuck you travis or whatever the hell your name is - recompile

Xoxeyos committed:
Oh god oh fuck clock cult cost 0 for a while, oh god oh fuck meteors are now in, only coming in at threat 100 and higher. Shadowling requirement shifted,

Xoxeyos committed:
recompile

Xoxeyos committed:
Rev Fix

Xoxeyos committed:
Allows more midrounds to play

Xoxeyos committed:
Clockwork additions and whatnot

Xoxeyos committed:
Adds Vampire, and some changes to the requirements, and whatnot.

Xoxeyos committed:
Fixed Jobs to Roles

Xoxeyos committed:
Adds Vampire to Late Join

Xoxeyos committed:
Adds vampire cooldown to roundstart

Xoxeyos committed:
Adds Traitor_Ruleset to Vampire

Xoxeyos committed:
Adds Vamps to midround

Xoxeyos committed:
recompile

Xoxeyos committed:
Shadowling Modification

Xoxeyos committed:
Revs are now under the flag TRAITOR_RULESET, shadowling change, attempts to keep the round from ending if all the shadowlings die.

Xoxeyos committed:
Makes mindshield limited stock, now costs 1 tc, very small chance to show up in surplus crate, isn't illegal tech, player mininum is 30

Xoxeyos committed:
Adds cargo crate discounts

Thank you Jamie and Gamer for helping!

Xoxeyos committed:
Adds the party crate to discount

Xoxeyos committed:
Adds Zombies to the midround

Xoxeyos committed:
Recompile

Xoxeyos committed:
FUCK YOU RECOMPILE ALREADY

Xoxeyos committed:
recompile

Xoxeyos committed:
Jungle Fever can now roll.

Xoxeyos committed:
Only need one vampire candidate now.

Xoxeyos committed:
Fix

Xoxeyos committed:
Ah shit it seems you're utterly fucked.

Xoxeyos committed:
Update dynamic_rulesets_roundstart.dm

If set to TRUE, dynamic mode will be able to draft this ruleset again later on. (doesn't apply for roundstart rules(?))

Xoxeyos committed:
No Highpop req for shadowling

Xoxeyos committed:
Adds Darkspawn

Xoxeyos committed:
Adds some logging.
